staging/src/k8s.io/kube-apiextensions-server/pkg/apiserver/customresource_handler.go

```diff
@@ -353,13 +360,38 @@ func (s UnstructuredNegotiatedSerializer) SupportedMediaTypes() []runtime.Serial
 }
 
 func (s UnstructuredNegotiatedSerializer) EncoderForVersion(serializer runtime.Encoder, gv runtime.GroupVersioner) runtime.Encoder {
-	return unstructured.UnstructuredJSONScheme
+	return versioning.NewDefaultingCodecForScheme(Scheme, crEncoderInstance, nil, gv, nil)
 }
 
 func (s UnstructuredNegotiatedSerializer) DecoderToVersion(serializer runtime.Decoder, gv runtime.GroupVersioner) runtime.Decoder {
 	return unstructured.UnstructuredJSONScheme
 }
 
+var crEncoderInstance = crEncoder{}
+
+// crEncoder *usually* encodes using the unstructured.UnstructuredJSONScheme, but if the type is one of the unversioned kinds
+// it will serialize them out
+type crEncoder struct{}
+
+func (crEncoder) Encode(obj runtime.Object, w io.Writer) error {
+	switch obj.(type) {
+	case
+		*metav1.Status,
+		*metav1.WatchEvent:
+
+		var serializer runtime.Serializer
+		for _, info := range Codecs.SupportedMediaTypes() {
```

Comment:
> having seen this loop at a couple of places. We should have a getter for a given media type.

I don't remember seeing it, but something in that negotiated serializer API has to change.  It's really difficult to sort out.  Feels like we're receiving information as input that we don't need.